fork_daemon.php

```diff
@@ -770,12 +773,15 @@ public function signal_handler_sighup($signal_number)
 		else
 		{
 			// child received sighup. note a child is only in one bucket, do not loop through all buckets
-			$this->log('child process [' . getmypid() . '] received sighup with bucket type [' . $this->child_bucket . ']', self::LOG_LEVEL_DEBUG);
-			$this->invoke_callback(
-				$this->child_function_sighup[$this->child_bucket],
-				array($this->child_bucket),
-				true
-			);
+			if (isset($this->child_bucket))
+			{
+				$this->log('child process [' . getmypid() . '] received sighup with bucket type [' . $this->child_bucket . ']', self::LOG_LEVEL_DEBUG);
+				$this->invoke_callback(
+					$this->child_function_sighup[$this->child_bucket],
```

Comment:
The if statement doesn't check for $this->child_function_sighup[$this->child_bucket]. Can we add that too?